src/Xamarin.Android.Tools.Bytecode/JavaParameterNamesLoader.cs

```diff
@@ -0,0 +1,192 @@
+
+					if (matched == null) {
+						Log.Warning (0, "Method {0}({1}) not found.",
+						             package + '.' + type + '.' + mname,
+						             string.Join (",", parameters.Select (para => para.Attribute ("type").Value)));
+						continue;
+					}
+
+					for (int i = 0; i < parameters.Length; i++)
+						parameters [i].SetAttributeValue ("name", matched.Parameters [i].Name);
+
+					matched.Applied = true;
+					break;
+				}
+			}
+			foreach (var p in fixup)
+				foreach (var t in p.Types)
+					foreach (var m in t.Methods.Where (m => !m.Applied))
+						Log.Warning (0, "Method parameter description for {0}.{1}.{2}({3}) is never applied",
+						             p.Name, t.Name, m.Name, string.Join (",", m.Parameters.Select (para => para.Type)));
+		}
+
+		class Parameter
+		{
+			public string Type { get; set; }
+			public string Name { get; set; }
+		}
+
+		class Method
+		{
+			public string Name { get; set; }
+			public List<Parameter> Parameters { get; set; }
+			public bool Applied { get; set; }
+		}
+
+		class Type
+		{
+			public string Name { get; set; }
+			public List<Method> Methods { get; set; }
+		}
+
+		class Package
+		{
+			public string Name { get; set; }
+			public List<Type> Types { get; set; }
+		}
+
+		// from https://github.com/atsushieno/xamarin-android-docimporter-ng/blob/83ed572ebfdcec8a8becd3337943a488bf56d57d/Xamarin.Android.Tools.JavaStubImporter/JavaApiParameterNamesXmlExporter.cs#L78
+		/*
+		 * The Text Format is:
+		 * 
+		 * package {packagename}
+		 * ---------------------------------------
+		 *   interface {interfacename}{optional_type_parameters} -or-
+		 *   class {classname}{optional_type_parameters}
+		 *     {optional_type_parameters}{methodname}({parameters})
+		 * 
+		 * optional_type_parameters: "" -or- "<A,B,C>" (no constraints allowed)
+		 * parameters: type1 p0, type2 p1 (pairs of {type} {name}, joined by ", ")
+		 * 
+		 * It is with strict indentations. two spaces for types, four spaces for methods.
+		 * 
+		 * Constructors are named as "#ctor".
+		 * 
+		 * Commas are used by both parameter types and parameter separators,
+		 * but only parameter separators can be followed by a whitespace.
+		 * It is useful when writing text parsers for this format.
+		 * 
+		 * Type names may contain whitespaces in case it is with generic constraints (e.g. "? extends FooBar"),
+		 * so when parsing a parameter type-name pair, the only trustworthy whitespace for tokenizing name is the *last* one.
+		 * 
+		 */
+		List<Package> LoadParameterFixupDescription (string path)
+		{
+			var fixup = new List<Package> ();
+			string package = null;
+			var types = new List<Type> ();
+			string type = null;
+			var methods = new List<Method> ();
+			foreach (var l in File.ReadAllLines (path)) {
+				if (l.Trim ().Length == 0 || l.StartsWith ("--------", StringComparison.Ordinal))
```

Comment:
Should this file format support *comments*, e.g. ignore all lines starting with `#`? I imagine that would be useful.

Reply:
Good idea.